src/index.ts

```
@@ -1,15 +1,21 @@
-//@ts-nocheck
+interface Person {
+  readonly id?: string
+  readonly name: string
+  readonly age?: number
+  readonly favoriteColor?: string
+}
 
 export default class UserAPI {
-  constructor(users) {
+  list: Array<Person>
+  constructor(users: Array<Person>) {
     this.list = users || []
   }
-  //random id generator
-  randomId = () => {
+
+  randomId = (): string => {
     return Math.random().toString(24).slice(2)
   }
 
-  addUser(user) {
+  addUser(user: Person): Person | Error {
```

error is never returned, so its not appropriate to declare this